File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.cpp
class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        int n = gain.size();
        int highest = 0;

        vector<int> answer(n+1 , 0);

        for(int i = 1 ; i< n+1  ;i++){
            answer[i] = answer[i-1] + gain[i-1];
            highest = max(highest, answer[i]);    
        }
        return highest;
    }
};

File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.py
class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        ans = 0
        alt_sum = 0
        for i in range(len(gain)):
            alt_sum += gain[i]

            if(ans < alt_sum):
                ans = alt_sum
        
        return ans